src/site/shared/containers/ItemNav/index.tsx

```diff
@@ -47,12 +55,27 @@ export const ItemNav = ({ info, config }: Props) => {
     // State to keep track of the number of times an item is clicked
     //  in relation to https://github.com/OpenCircuits/OpenCircuits/issues/579
     const [{curItemID, numClicks}, setState] = useState({curItemID: "", numClicks: 1});
-
+    const [hovering, setHover] = useState("")
     // Resets the curItemID and numClicks
     function reset() {
         setState({curItemID: "", numClicks: 1});
     }
+    function deleteIC(sec: ItemNavSection, ic: ItemNavItem) {
+        var designer: DigitalCircuitDesigner = info.designer as DigitalCircuitDesigner;
+        var shouldDelete: boolean = true;
 
+        info.designer.getAll().forEach(function (o) {
+            if (o instanceof IC && o.getData() === ic.data){
+                window.alert("Cannot delete this IC while instances remain in the circuit.");
+                shouldDelete = false;
+            }
+        })
+        if(shouldDelete){
+            sec.items.splice(sec.items.indexOf(ic));
+            info.history.add(new DeleteICDataAction(ic.data, designer).execute());
+            setHover("");
+        }
+    }
```

Comment:
So as stated above, shared shouldn't know that digital exists. 
So to fix this, what we can do is have `onDelete` as a function in the `Props` for `ItemNav` and then in `DigitalItemNav` you can pass in `onDelete` as this function